### PR TITLE
[Refactor] Replace npu_ring_mla with FIA in MLA prefill

### PR DESCRIPTION
**Pull request by LICO1314**
### What this PR does / why we need it?

**Refactor: Replace npu_ring_mla with FIA in MLA prefill**

This PR refactors the MLA (Multi-Layer Attention) prefill implementation by replacing `npu_ring_mla` with `npu_fused_infer_attention_score` (FIA) operator, unifying the attention backend with the standard attention implementation.

**Key changes:**

1. **Core prefill refactoring (`mla_v1.py`)**
   - Replace `npu_ring_mla` with `npu_fused_infer_attention_score` in `_forward_prefill` and `_compute_prefill_context`
   - Use TND layout with `softmax_lse_flag=True` for prefill attention
   - Use `npu_attention_update` to merge multiple chunk outputs with LSE (Log-Sum-Exp)
   - Change `attn_mask` from `get_final_mla_mask()` to `get_splitfuse_attn_mask()` for FIA compatibility

2. **Data type handling**
   - Add automatic float16 → bfloat16 conversion (FIA with TND layout only supports bfloat16)
   - Convert output back to original dtype after FIA computation

3. **Metadata optimization**
   - Pre-calculate `actual_seq_lengths_q` in `AscendMLAPrefillMetadata`
   - Pre-calculate `chunk_actual_seq_lengths_kv_list` in `ChunkedContextMetadata`
   - Move `torch.cumsum` operations from forward pass to metadata building phase

4. **CP compatibility (`mla_cp.py`)**
   - Add `_ring_mla_mask_builder` to get `npu_ring_mla`-compatible masks for Context Parallel scenarios
   - Add `chunk_actual_seq_lengths_kv_list` field to `CPChunkedContextMetadata`

**Why we need it:**
- **Backend unification**: Aligns MLA prefill with standard attention implementation (`attention_v1.py`)
- **Better chunked context support**: FIA + `npu_attention_update` provides native LSE-based output merging
- **Future compatibility**: Prepares for eventual `npu_ring_mla` removal across the codebase

### Does this PR introduce _any_ user-facing change?

**No.** This is a pure refactoring with no functional changes - same behavior, unified backend.

---
- Related issue: #5463 (item 7)
- vLLM version: v0.14.1

### CODE REVIEW

**Comment by jianzs** [comment]
How much performance improvement does this change bring? If it's not significant, I suggest waiting until FIA supports chunk prefill before making this modification. Otherwise, this PR won't achieve your goal and will only add unnecessary code complexity.

File: vllm_ascend/attention/mla_v1.py [hunk 1]
```diff
@@ -975,19 +976,23 @@ def _compute_prefill_context(
...
-                                       dtype=torch.int32)
         cache_kv_c = kv_c_and_k_pe_cache[0]
         cache_k_pe = kv_c_and_k_pe_cache[1]
         num_heads = cache_k_pe.size(2)
         latent_kv_dim = kv_c_and_k_pe_cache[0].size(-1)
+
+        query_lens = prefill_metadata.query_lens
+        if isinstance(query_lens, list):
+            query_lens = torch.tensor(query_lens)
+        actual_seq_lengths_q = torch.cumsum(query_lens, dim=0).tolist()
```
Comment (weijinqian0) [inline vllm_ascend/attention/mla_v1.py]: Why add this?  This parameter can be added in AscendMlaMetadataBuilder like decode_metdata.

**Review by wangxiyuan — APPROVED** [review]

**Comment by jianzs** [comment]
@LICO1314  Please rebase your code
